test/ptx_tests.jl

```diff
@@ -504,4 +504,25 @@ end
     ir = sprint(io->PTX.code_llvm(io, kernel_inline, Tuple{Ptr{Int64}, Int64}, meta=Plugin.NeverInlineMeta()))
     @test occursin("call fastcc i64 @julia_inline", ir)
 end
+
+@testset "Mock Enzyme" begin
+    function f(x)
+        x^2
+    end
+    
+    function kernel(a, x)
+        y = Plugin.autodiff(f, x)
+        unsafe_store!(a, y)
+        nothing
+    end
+    
+
+    @show PTX.code_typed(kernel, Tuple{Ptr{Float64}, Float64}, meta=Plugin.MockEnzymeMeta())
+
+    # FIXME: the fact that meta is necessary here almost invalidates that extension mechanism
+    #        we somehow need to be able to add this kind of "autodiff" abs int handling.
+    ir = sprint(io->PTX.code_llvm(io, kernel, Tuple{Ptr{Float64}, Float64}, meta=Plugin.MockEnzymeMeta()))
```

Comment:
Grrmbl, I need to rethink #634 since I somehow need to tell CUDA version of GPUCompiler what to do when it sees a call to `autodiff`